sympy/crypto/crypto.py

```diff
@@ -1383,24 +1396,227 @@ def bifid6_square(key=None):
+            new_primes.extend(factorint(i, multiple=True))
+        primes = new_primes
+
+    n = reduce(lambda i, j: i*j, primes)
+
+    tally = multiset(primes)
+    if all(v == 1 for v in tally.values()):
+        multiple = list(tally.keys())
+
+        if totient == 'Euler':
+            from sympy.ntheory import totient as totfunc
+            phi = totfunc._from_distinct_primes(*multiple)
+
+        elif totient == 'Carmichael':
+            from sympy.ntheory import reduced_totient as totfunc
+            phi = totfunc._from_distinct_primes(*multiple)
+
+    else:
+        if not multipower:
+            NonInvertibleCipherWarning(
+                'Non-distinctive primes found in the factors {}. '
+                'The cipher may not be decryptable for some numbers '
+                'in the complete residue system Z[{}], but the cipher '
+                'can still be valid if you restrict the domain to be '
+                'the reduced residue system Z*[{}]. You can pass '
+                'the flag multipower=True if you want to suppress this '
+                'warning.'
+                .format(primes, n, n)
+                ).warn()
+
+        if totient == 'Euler':
+            from sympy.ntheory import totient as totfunc
+            phi = totfunc._from_factors(tally)
+
+        elif totient == 'Carmichael':
+            from sympy.ntheory import reduced_totient as totfunc
+            phi = totfunc._from_factors(tally)
+
+    if igcd(e, phi) == 1:
+        if public and not private:
+            return n, e
+
+        if private and not public:
+            d = mod_inverse(e, phi)
+            return n, d
+
+    return False
+
+
+def rsa_public_key(*args, **kwargs):
+    r"""Return the RSA *public key* pair, `(n, e)`
+
+    Parameters
+    ==========
+
+    args : naturals
+        If specified as `p, q, e` where `p` and `q` are distinct primes
+        and `e` is a desired public exponent of the RSA, `n = p q` and
+        `e` will be verified against the totient
+        `\phi(n)` (Euler totient) or `\lambda(n)` (Carmichael totient)
+        to be `\gcd(e, \phi(n)) = 1` or `\gcd(e, \lambda(n)) = 1`.
+
+        If specified as `p_1, p_2, ..., p_n, e` where
+        `p_1, p_2, ..., p_n` are specified as primes,
+        and `e` is specified as a desired public exponent of the RSA,
+        it will be able to form a multi-prime RSA, which is a more
+        generalized form of the popular 2-prime RSA.
+
+        It can also be possible to form a single-prime RSA by specifying
+        the argument as `p, e`, which can be considered a trivial case
+        of a multiprime RSA.
+
+        Furthermore, it can be possible to form a multi-power RSA by
+        specifying two or more pairs of the primes to be same.
+        However, unlike the two-distinct prime RSA or multi-prime
+        RSA, not every numbers in the complete residue system
+        (`\mathbb{Z}_n`) will be decryptable since the mapping will not
```

Comment:
What is the mapping that is referred to here?

Reply:
It's referring to complete residue system to itself for multiprime RSA cases, and reduced residue system to itself for multipower RSA cases